crates/cli-support/src/js/rust2js.rs

```diff
@@ -528,10 +543,31 @@ impl<'a, 'b> Rust2Js<'a, 'b> {
         ret.push_str(") {\n");
         ret.push_str(&self.prelude);
 
-        let mut invoc = self.ret_expr.replace(
-            "JS",
-            &format!("{}({})", invoc, self.js_arguments.join(", ")),
-        );
+        let mut invoc = if self.variadic {
+            if self.js_arguments.is_empty() {
+                unreachable!("the last argument of a variadic must be a slice");
+            }
+            let last_arg = self.js_arguments.len() - 1; // check implies >= 0
+            // The variadic part of the arguments could be passed as as a typed array in
+            // javascript, so we cannot just do `.concat(..)`, as this seems to convert the
+            // contents to strings (javascript is very strange).
+            write!(ret, "\
+                let args = [{}];\n\
+                {}.forEach(extra_arg => args.push(extra_arg));\n\
```

Comment:
We're already using pretty modern syntax elsewhere, so instead of this could it instead generate `foo(a, b, ...c)` perhaps?

Reply:
This seems to work - I was expecting the problem mentioned in the comment.

I'll add plenty of tests to try with lots of different types.

Reply:
done